airflow/providers/apache/hive/hooks/hive.py

```diff
@@ -708,7 +708,11 @@ def _get_max_partition_from_part_specs(part_specs, partition_key, filter_map):
         if not candidates:
             return None
         else:
-            return max(candidates).encode('utf-8')
+            max_val = max(candidates).encode('utf-8')
+            # this check is for python3 to keep same behavior of returning a string type
+            if isinstance(max_val, (bytes, bytearray)):
```

Comment:
this seems very odd -we're encoding to just decode it in the next step??

Reply:
on master only python 3 is supported - you can probably tidy this up as a result

Reply:
Oh if that is the case then let me clean this up and redo this. Thank you @ashb !

Reply:
@ashb went ahead and just gutted any call to encode it. since we're all in py3 our response is a string as it is, and we don't need to encode to bytes since we want it decoded for use